woocommerce/woocommerce-android
# [Scan to update inventory] Part 2 – Product update

#### [Scan to update inventory] Part 2 – Product's quantity update, unit tests & edge cases
This is **Part 2** of the Scan to update inventory project
Part 1 – https://github.com/woocommerce/woocommerce-android/pull/10250
Part 3 – https://github.com/woocommerce/woocommerce-android/pull/10278
Part 4 – https://github.com/woocommerce/woocommerce-android/pull/10313

### Description

Designs: w3JBReW1UJesCf02sj43kR-fi-775%3A34738

Changes:
1. Product stock quantity update
2. Showing error snackbars:
- when product is not stock-managed
- when product is not found for scanned sku
- when there was error during update operation
3. Unit tests for `FetchProductBySKU` and `ScanToUpdateInventoryViewModel`

### Testing instructions

1. Open product list
2. Click "barcode scanner" icon
3. Scan barcode with the SKU of existing product (you can use online barcode generator like [this one](https://barcode.tec-it.com/en/?data=ACGADO))
4. Verify product is found and the bottom sheet is opened
5. Click Quantity + 1 and verify the product quantity is updated successfully

In case of scanning the SKU of a product that is not stock-managed there should be a snack bar information shown.

### Images/gif

https://github.com/woocommerce/woocommerce-android/assets/4527432/2e8801c1-5b7f-410b-814f-d4a0d4af4ce7
